dpytools/http/upload.py

```diff
@@ -0,0 +1,131 @@
+import datetime
+import os
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Tuple, Union
+
+from dpytools.http.base import BaseHttpClient
+
+
+class UploadClient(BaseHttpClient):
+    def __init__(self, upload_url: str):
+        # Inherit backoff_max value from BaseHTTPClient.__init__
+        super().__init__()
+        self.upload_url = upload_url
+        # For security purposes, `self.access_token` should be set using a FLORENCE_TOKEN environment variable
+        self.access_token = os.environ.get("FLORENCE_TOKEN")
+        self.headers = {"X-Florence-Token": self.access_token}
+
+    # TODO The `upload` method is currently configured to match the old `upload` endpoint specification. Once the `upload-new` endpoint has been released, the `params` argument passed to `self.post` will need to be adjusted to the new specification (see https://github.com/ONSdigital/dp-upload-service/blob/b91b53a13b28c6ba1095abee71680c591862c68c/swagger.yaml).
+    def upload(
+        self,
+        csv_path: Union[Path, str],
+        s3_path: str,
+        chunk_size: int = 5242880,
+    ) -> Tuple[str, str]:
+        """
+        Upload files to the DP Upload Service. Files (located at `csv_path`) are chunked (default chunk size 5242880 bytes) and uploaded to the S3 bucket located at `s3_path`.
+
+        Returns the S3 Object key and full S3 URL of the uploaded file.
+        """
+        # Get total size of file to be uploaded
+        total_size = str(os.path.getsize(csv_path))
+
+        # Convert csv_path string to Path
+        if not isinstance(csv_path, Path):
+            csv_path = Path(csv_path).absolute()
+
+        # Get filename from csv filepath
+        filename = str(csv_path).split("/")[-1]
+
+        # Get timestamp to create `resumableIdentifier` value in `POST` params
+        timestamp = datetime.datetime.now().strftime("%d%m%y%H%M%S")
+
+        # Create file chunks
+        file_chunks = self._create_temp_chunks(csv_path=csv_path, chunk_size=chunk_size)
+
+        chunk_number = 1
+
+        for file_chunk in file_chunks:
+            with open(file_chunk, "rb") as f:
+                # Load file chunk as binary data
+                file = {"file": f}
+
+                # TODO current_chunk_size = len(f.read()) is causing the request to fail?
```

Comment:
solve or remove the reference please, if we're happy with setting the chunk size then this can go.